text/3759-cargo-supported-targets.md

```diff
@@ -0,0 +1,494 @@
+Other formats can be considered:
+
+Using a list of `cfg` strings, and also accepting explicit target-triples:
+```toml
+supported-targets = [
+    'cfg(target_family = "unix")',
+    'cfg(target_family = "wasm")',
+    "x86_64-pc-windows-gnu",
+]
+```
+This can be unintuitive to understand however, as the list implies a union of all its elements,
+which is not immediately obvious.
+
+Using the `[target]` table, for example:
+```toml
+[target.'cfg(target_os = "linux")']
+supported = true
+```
+If the list of supported targets is long (should it ever be?), then the `Cargo.toml` file becomes
+very verbose as well.
+
+A `[suppported]` table, with `arch = ["<arch>", ...]`, `os = ["<os>", ...]`, `target = ["<target>",
+...]`, etc. This is more verbose, complex to implement, learn, and remember. It is also not obvious
+how `not` and `all` could be represented in this format. For example:
+```toml
+[supported]
+os = ["linux", "macos"]
+arch = ["x86_64"]
+```
+
+## Naming
+
+Some other names for this field can be considered:
+
+- `required-targets`. Pro: it matches with the naming of `required-features`. Con:
+  `required-features` is a list of features that must _all_ be enabled (conjunction), whereas
+  `supported-targets` is a list of targets where _any_ is allowed (disjunction).
+- `targets`. As in "this package _targets_ ...". Pro: Concise. Con: Ambiguous, and could be confused
+  with the `target` table.
+
+## Package scope vs. cargo-target scope
+
+The `supported-targets` field is placed at the package level, and not at the cargo-target level
+(i.e., under, `[lib]`, `[[bin]]`, etc.)
+
+It is possible to allow cargo-targets to further restrict the `supported-targets` of the package,
+but this is left as a [future possibility](#future-possibilities).
+
+See also: [using a package vs. using a workspace](package-vs-workspace).
+
+[package-vs-workspace]:
+#https://blog.rust-lang.org/inside-rust/2024/02/13/this-development-cycle-in-cargo-1-77.html#when-to-use-packages-or-workspaces
+
+## Not using `cfg`
+
+Using `cfg` complicates the implementation (and thus maintenance), and may require a substantial
+amount of calls to `rustc` to check target-`cfg` compatibility. Some alternatives are discussed here
+along with their drawbacks.
+
+### Using wildcards
+
+Instead of using `cfg` specifications, one could use wildcards (e.g., `x86_64-*-linux-*`). This is
+much simpler to implement, target-triples are syntactically checked for a match instead of solving
+set relations for `cfg`. However, this is not as expressive as `cfg`, and does not correctly
+represent the semantics of target-triples. For example, supporting `target_family = "unix"` would
+require an annoyingly long list of wildcard patterns. Things like `target_pointer_width = "32"` are
+even harder to represent, and things like `target_feature = "avx"` are basically not representable.
+Also, this is new syntax not currently used by cargo.
+
+### Allowing only target-triples
```

Comment:
As target-tuples were removed, we should also have section talking about including them